lib/cpp/Kconfig

```
@@ -12,6 +12,26 @@ config CPP
 
 if CPP
 
+config STD_CPP_VERSION
+	int
+	default 1998 if STD_CPP98
+	default 2011 if STD_CPP11
+	default 2014 if STD_CPP14
+	default 2017 if STD_CPP17
+	default 2020 if STD_CPP20 || STD_CPP2A || STD_CPP2B
+	help
+		The version number of C++ standard being used (NOTE: this uses the full
+		year). This config can be used to check for a minimum supported
+		version. Example:
+
+		  depends on STD_CPP_VERSION >= 2017
+
+		Adding this to your library's enablement Kconfig will force a minimum c++17
+		standard.
+
+		The full year is used so c++98 can be checked using > and < operators
+		without conflicting with c++11 or higher.
```

You need a single tab + 2 spaces for the `help` indentation.

```suggestion
	  The version number of C++ standard being used (NOTE: this uses the full
	  year). This config can be used to check for a minimum supported
	  version. Example:

	    depends on STD_CPP_VERSION >= 2017

	  Adding this to your library's enablement Kconfig will force a minimum c++17
	  standard.

	  The full year is used so c++98 can be checked using > and < operators
	  without conflicting with c++11 or higher.
```

---

Thanks